### PR TITLE
Add `is_test` for daemons to use temporary files 

### PR DESCRIPTION
**Pull request by Buckram123**
Randomly generated name for temporary file inside tempdir of OS will be used when `is_test` provided for daemons, this file will be removed by OS when no longer needed

### Checklist

- [x] Changelog updated.
- [ ] Docs updated.

### CODE REVIEW

File: cw-orch-daemon/src/state.rs [hunk 1]
```diff
@@ -331,6 +332,19 @@ impl StateInterface for DaemonState {
     }
 }
 
+// copied from `tempfile::util::tmpname` implementation
+pub(crate) fn gen_temp_file_path() -> std::path::PathBuf {
```
Comment (CyberHoward) [inline cw-orch-daemon/src/state.rs:336]: We use uid and tempfile crates in node tests

https://docs.rs/uid/latest/uid/
https://docs.rs/tempfile/latest/tempfile/

Do you prefer this setup?

File: cw-orch-daemon/tests/daemon_state.rs [hunk 2]
```diff
@@ -216,22 +216,22 @@ fn reuse_same_state_multichain() {
...
-            std::thread::sleep(std::time::Duration::from_millis(50));
-            let _state = JsonLockedState::new(TEST_STATE_FILE);
-        }
-        Err(_) => (),
-    }
-}
+// #[test]
+// #[serial_test::serial]
+// #[should_panic]
+// #[ignore = "Serial don't track forks for some reason, run it manually"]
```
Comment (CyberHoward) [inline cw-orch-daemon/tests/daemon_state.rs:222]: Can we update this test or why is it commented?

File: cw-orch-daemon/src/state.rs (hunk 1 above)
Comment (Buckram123) [reply cw-orch-daemon/src/state.rs:336]: Yeah uid is be nice, thanks!
I'm not sure if we can use tempfile crate as it returns `File` object, which means we don't have control of creation for this object, but we need it for locking file, beyond that it's pretty hacky to get path from the `File` object

Comment (Buckram123) [reply cw-orch-daemon/src/state.rs]: Done, also noticed that uid not used (anymore?) in node tests, so removed unused part from there as well!

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**